OpenRA.Mods.Common/Traits/BotModules/BaseBuilderBotModule.cs

```diff
@@ -231,21 +233,47 @@ void SetRallyPointsForNewProductionBuildings(IBot bot)
 				if (rp.Actor.Owner != player)
 					continue;
 
-				if (rp.Trait.Path.Count == 0 || !IsRallyPointValid(rp.Trait.Path[0], rp.Actor.Info.TraitInfoOrDefault<BuildingInfo>()))
+				var production = rp.Actor.TraitsImplementing<Production>().FirstOrDefault();
+				if (production == null)
+					continue;
+
+				var queue = AIUtils.FindQueues(player, production.Info.Produces.First()).FirstOrDefault();
+				if (queue == null)
+					continue;
+
+				var buildableThings = queue.BuildableItems();
+				if (!buildableThings.Any())
+					continue;
+
+				var mobileInfos = buildableThings.Select(s => s.TraitInfoOrDefault<MobileInfo>());
+				foreach (var mobileInfo in mobileInfos)
 				{
-					bot.QueueOrder(new Order("SetRallyPoint", rp.Actor, Target.FromCell(world, ChooseRallyLocationNear(rp.Actor)), false)
+					if (mobileInfo == null)
+						continue;
+
+					if (rp.Trait.Path.Count == 0 || !IsRallyPointValid(rp.Trait.Path[0], rp.Actor, mobileInfo))
```

Comment:
And then remove the first part of this check here.

Reply:
That would crash as `rp.Trait.Path[0]` requires at least one path set.